pkg/controller/networkpolicy/validate.go

```diff
@@ -285,17 +285,27 @@ func (v *antreaPolicyValidator) validatePort(ingress, egress []crdv1alpha1.Rule)
 	isValid := func(rules []crdv1alpha1.Rule) error {
 		for _, rule := range rules {
 			for _, port := range rule.Ports {
-				if port.EndPort == nil {
+				if port.EndPort == nil && port.SourceEndPort == nil {
 					continue
 				}
```

Comment:
nit: the 3 lines don't change the result but make each of the two conditions checked twice in some cases, can be safely removed.